Source/WebCore/Modules/applepay/ApplePaySession.cpp

```diff
@@ -64,6 +65,7 @@
 #include "SecurityOrigin.h"
 #include "Settings.h"
 #include "UserGestureIndicator.h"
+#include <CoreFoundation/CFTimeZone.h>
```

Comment:
It seems wrong to include a `CoreFoundation` header at this layer of the code, even though `ENABLE(APPLE_PAY)` is only true on Cocoa platforms.

It might be better to add a helper in WTF, using icu. Something like `canonicalizeTimeZoneName` could be moved from JSC into WTF.

Thoughts @hortont424?

Reply:
Unless it's a trivial change, I'd appreciate if we could defer this change to convergence (I'd file a radar and add a FIXME). Please let me know.

Reply:
I think it would just be moving (re-using) the logic in `canonicalizeTimeZoneName`. My understanding is that if that method returns a `null` String, the time zone is invalid.

Maybe @Constellation has thoughts about where the helper could live?

Reply:
In this latest update, I've just copied the `canonicalizeTimeZoneName` code where I needed it, with a FIXME to later move it to WTF, because I think it would probably need some careful consideration and testing. (Happy to open a radar if you agree.)
But please let me know if you'd prefer another location or solution right now.

Reply:
New update, now using `WTF::isTimeZoneValid`, thank you for the hint.